### DISCUSSION TITLE
Creating schedules with Ajax

### DISCUSSION THREAD

**Discussion by RJpedro** [nhn/tui.calendar#1073] [Q&A] [answered]
Hello, my name is João Pedro. First of all my team and I would like to thank you immensely for this beautiful library... THANK YOU SO MUCH.

I'm currently trying to create schedules inside my AJAX, and it's not working. I'm pulling data from the database and storing it inside an Object, I do it through a for, inside that for I create the schedule body following the model provided by you. However, when I give the command "createSchedules()" passing the object I created but unfortunately nothing happens.

**Comment by adhrinae** [comment, marked as answer]
Hmm... It's exactly how this library is used in other products of this corporation.
Can you upload a code that is able to reproduce the error? I cannot understand what is wrong at this moment.

**Reply by RJpedro** [reply]
I thank the attention. I will send you the code so you can evaluate. I'm still new to programming, so forgive me if it's a stupid mistake.

**Reply by RJpedro** [reply]
var calendar = new tui.Calendar('#calendar', {
            isReadOnly: true,
            showToday: true,
            template: {
                alldayTitle: function() {
                    return '<span class="tui-full-calendar-left-content" style="text-align: center"> DIA INTEIRO </span>';
                },
                popupDetailBody: function(schedule) {
                    return '<strong>Jorge</strong> : ' + schedule.body;
                },
            },
            scheduleView: 'time',
            backgroundColor: '#b4b',
            defaultView: 'month',
            useDetailPopup: true,
            month: {
                daynames: ['Dom', 'Seg', 'Ter', 'Qua', 'Qui', 'Sex', 'Sab'],
                startDayOfWeek: 1,
                narrowWeekend: false,
                moreLayerSize: {
                    width: '500px',
                },
                grid: {
                    header: {
                        height: 34
                    },
                    footer: {
                        height: 10
                    }
                },
            },
        });
        var idUnidade = "<?php echo $sessionUsuarioUnidade; ?>"
        var action = "../controller/NovoCalendarioController.php?action=consultarAgendamentosInRange2JSON";
        action += "&unidade=" + idUnidade;
        var jqxhr = $.ajax({
            url: action,
            type: 'post',
            enctype: 'multipart/form-data',
            processData: false,
            contentType: false,
        }).then(
            function(data) {
                data = JSON.parse(data);
                var status = data.success;
                var dados = data.result;
                if (status == 1 && dados.length > 0) {
                    for (var count in dados) {
                        schedule = {
                            id: dados[count].id,
                            calendarId: dados[count].id,
                            body: dados[count].body,
                            title: dados[count].body,
                            start: dados[count].start,
                            end: dados[count].end,
                            borderColor: dados[count].borderColor,
                            bgColor: dados[count].bgColor,
                        }
                        console.log(schedule)
                        calendar.createSchedules(schedule, true);
                    }
                }
            }
        ).fail(
            function() {
                console.log("error");
            }
        );
        // calendar.createSchedules(schedule, true);
        calendar.toggleTaskView(false);
        calendar.render(false);

**Reply by adhrinae** [reply]
the `createSchedules` method takes an array as its first argument for taking multiple schedules at once. but you're passing the schedule itself.

```js
// change this
calendar.createSchedules(schedule);

// to this
calendar.createSchedules([schedule]);
```

**Reply by RJpedro** [reply]
You are sure. I made this adjustment and it kept going wrong because I was forgetting to declare the appointment category.
Now everything is working fine, scheduling correctly. Thanks again immensely for this amazing library and help given.
